### PR TITLE
Add events for GunComponent values, muzzle flashes and cartridge spread

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
This also lets GunComponent and other components modify how many shorts are fired per burst.

## Media
https://github.com/space-wizards/space-station-14/assets/10968691/c96639db-512e-47c1-937a-a74561f41c28

https://github.com/space-wizards/space-station-14/assets/10968691/27718ca2-c321-4ce5-9b56-e46fb7be404c

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking Changes
- GunComponent values can now only be updated by GunSystem. Use the new GunRefreshModifiersEvent to change them from other systems.

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
By chance, does this solve the issue where you can avoid the burst mode being different from full auto by just clicking really fast?

**Comment by DrSmugleaf** [comment]
> By chance, does this solve the issue where you can avoid the burst mode being different from full auto by just clicking really fast?

Don't think so

File: Content.Server/PneumaticCannon/PneumaticCannonSystem.cs [hunk 1]
```diff
@@ -105,6 +108,11 @@ private void OnShoot(Entity<PneumaticCannonComponent> cannon, ref GunShotEvent a
         _slots.TryEject(uid, PneumaticCannonComponent.TankSlotId, args.User, out _);
     }
 
+    private void OnGunRefreshModifiers(Entity<PneumaticCannonComponent> ent, ref GunRefreshModifiersEvent args)
+    {
+
+    }
+
```
Comment (metalgearsloth) [inline Content.Server/PneumaticCannon/PneumaticCannonSystem.cs:115]: Head empty?

**Review by metalgearsloth — APPROVED** [review]

File: Content.Server/PneumaticCannon/PneumaticCannonSystem.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Server/PneumaticCannon/PneumaticCannonSystem.cs]: Forgot to add that part of the diff to the commit, fixed in [`0b6ded7` (#24077)](https://github.com/space-wizards/space-station-14/pull/24077/commits/0b6ded789c63ccc5b9a796fdd024feb35b5d6da8)

**Pull request merged**